src/net/java/sip/communicator/impl/protocol/jabber/ProtocolProviderServiceJabberImpl.java

```diff
@@ -1114,8 +1114,33 @@ private ConnectState connectAndLogin(
         confConn.setXmppDomain(serviceName);
         if (isBosh)
         {
-            confConn.setHost(boshURL);
-            // FIXME use setHostAddress, use proxy
+            BOSHConfiguration.Builder boshConfConnBuilder =
+                (BOSHConfiguration.Builder)confConn;
+
+            try
+            {
+                URI boshURI = new URI(boshURL);
+                boolean useHttps = boshURI.getScheme().equals("https");
+
+                int port = boshURI.getPort();
+                if (port == -1)
+                {
+                    port = useHttps ? 443 : 80;
+                }
+
+                boshConfConnBuilder
+                    .setUseHttps(useHttps)
+                    .setFile(boshURI.getPath() + "?" + boshURI.getQuery())
```

Comment:
Is there always a query? I know it works with an empty string after the ?, but it's ugly.

Reply:
I've updated and forced pushed the last commit. Does it look ok now @ibauersachs? Thanks.